game-core/src/main/java/games/strategy/engine/data/GameData.java

```diff
@@ -215,8 +213,19 @@ public UnitTypeList getUnitTypeList() {
     return unitTypeList;
   }
 
-  public DelegateList getDelegateList() {
-    return delegateList;
+  public Collection<IDelegate> getDelegates() {
```

Comment:
I think there may be a code-smell here, clients iterating over a list of delegates seems pretty dirty. I support the update to go from `DelegateList` to `Collection`, any objection if we mark this method as deprecated so we know to avoid using it?

Reply:
Well this public method is being used to store delegate information in the savegame, so it's probably necessary at some point?
Not sure why IDelegates aren't getting serialized directly like everything else, just some general information in string form that is sufficient to re-create the delegate on launch. I asked veqryn, but he couldn't tell me either.

Reply:
As far as I can tell, this API is just another version of `getDelegate(String)`, but with a complete lack of ownership pushing that to the client, eg:

(DefaultPlayerBridge:68)
```
        final IDelegate delegate = game.getData().getDelegateList().getDelegate(currentDelegate);

```

The client could just as well call:
```
game.getData().getDelegate(currentDelegate);
```

I suspect this is mostly a bad API and a further breaking of encapsulation.

Reply:
Well since `DelegateList` doesn't exist anymore in this PR, the former case doesn't work anymore, so `getDelegate(String)` is the only way to efficiently lookup `IDelegate`s by name.
`GameData#getDelegates` however is only used in for-each loops to serialize information or to generate xmls based off that information. having to call every delegate explicitly wouldn't make things more readable.